component/azstorage/block_blob.go

```diff
@@ -654,6 +656,197 @@ func (bb *BlockBlob) WriteFromBuffer(name string, metadata map[string]string, da
+}
+
+// create our definition of block
+func (bb *BlockBlob) createBlock(blockIdLength, startIndex, size int64) *common.Block {
+	newBlockId := base64.StdEncoding.EncodeToString(common.NewUUID(blockIdLength))
+	newBlock := &common.Block{
+		Id:         newBlockId,
+		StartIndex: startIndex,
+		EndIndex:   startIndex + size,
+		Size:       size,
+	}
+	return newBlock
+}
+
+func (bb *BlockBlob) createNewBlocks(modBlockList common.BlockOffsetList, blockList common.BlockOffsetList, offset int64, length int64, blockIdLength int64) (bool, int64, common.BlockOffsetList, common.BlockOffsetList) {
+	// BufferSize is the size of the buffer that will go beyond our current blob (appended)
+	var bufferSize int64
+	// counter will help us keep track of how many blocks we've created
+	counter := int64(0)
+	// appendOnly means there is no overlap at all with the blob - data only being appended
+	appendOnly := false
+	// if modBlockList is not of size 0 then we know that we're overwriting part of our current data - with new data also being appended to the end
+	if len(modBlockList) != 0 {
+		bufferSize = length - (blockList[len(blockList)-1].EndIndex - offset)
+		// only new data at the end basically offset => lastBlock.EndIndex
+	} else {
+		bufferSize = (offset + length) - blockList[len(blockList)-1].EndIndex
+		appendOnly = true
+	}
+	// reset the offset to be the last index in the block list since we are adding new blocks
+	offset = blockList[len(blockList)-1].EndIndex
+	startIndex := offset
+	for i := 0; i < int(bufferSize); i++ {
+		counter += 1
+		// create a new block if we hit our block size
+		if int64(i)%bb.Config.blockSize == 0 && i != 0 {
+			newBlock := bb.createBlock(blockIdLength, startIndex, bb.Config.blockSize)
+			modBlockList = append(modBlockList, newBlock)
+			blockList = append(blockList, newBlock)
+			startIndex = newBlock.EndIndex
+			// reset the counter since it will help us to determine if there is leftovers at the end
+			counter = 0
+		}
+	}
+	if counter != 0 {
+		newBlock := bb.createBlock(blockIdLength, startIndex, counter)
+		modBlockList = append(modBlockList, newBlock)
+		blockList = append(blockList, newBlock)
+	}
+	return appendOnly, bufferSize, modBlockList, blockList
+}
+
+// Write : write data at given offset to a blob
+func (bb *BlockBlob) Write(name string, offset int64, length int64, data []byte, FileOffsets common.BlockOffsetList) error {
+	log.Trace("BlockBlob::Write : name %s offset %v", name, offset)
+	defer log.TimeTrack(time.Now(), "BlockBlob::Write", name)
+	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
+
+	// does the file consist of blocks?
+	multipleBlocks := true
+	// tracks the case where our offset is great than our current file size (appending only - not modifying pre-existing data)
+	appendOnly := false
+	var blockList common.BlockOffsetList
+	var dataBuffer *[]byte
+
+	// if this is not 0 then we passed a cached block ID list - from stream for example
```

Comment:
instead of saying something like, "if this is not 0", which is descriptive from the code, could you add like when this is 0, this means this